pymeasure/instruments/keithley/keithley2281S.py

```diff
@@ -0,0 +1,573 @@
+            upper_voltage (float): charge end voltage
+        """
+        self.write(f":BATT:TEST:SENS:AH:GMOD:RANG {lower_voltage}, {upper_voltage}")
+
+    def save_model_to_usb(self, memory_slot: int, model_file_name: str):
+        """Save battery model to USB
+
+        Args:
+            memory_slot (int): Number of memory slot to save model from. Valid values are 1 to 10.
+            model_file_name (str): Name of battery model to save. '.csv' will be appended by device
+
+        Raises:
+            ValueError: invalid memory slot given
+        """
+        if _INTERNAL_MEMORY_SLOTS[0] > memory_slot > _INTERNAL_MEMORY_SLOTS[-1]:
+            raise ValueError
+        self.write(f":BATT:MOD:SAVE:USB {memory_slot}, \"{model_file_name}\"")
+
+    @property
+    def buffer_data(self) -> pd.DataFrame:
+        """Get the buffer in battery test mode and return its content as a pandas dataframe"""
+        if not self.parent.reading_available:
+            return pd.DataFrame(
+                {"current": [], "voltage": [], "capacity": [], "resistance": [], "time": []}
+            )
+        data = np.array(self._query_buffer_data, dtype=np.float32)
+        return pd.DataFrame(
+            {
+                "voltage": data[0::5],
+                "current": data[1::5],
+                "resistance": data[2::5],
+                "capacity": data[3::5],
+                "time": data[4::5],
+            }
+        )
+
+
+class BatterySimulationChannel(Channel):
+    """
+    Battery Simulation Mode of the device
+
+    Make sure that the parent's instrument `function_mode` property is set to `SIMULATION` before
+    using this channel.
+    """
+
+    select_model_internal = Instrument.setting(
+        ":BATT:MOD:RCL %d",
+        """Set the battery model from internal memory.""",
+        validator=strict_discrete_set,
+        values=_INTERNAL_MEMORY_SLOTS,
+    )
+
+    def load_model_from_usb(self, memory_slot: int, model_file_name: str):
+        """Load battery model from USB
+
+        Args:
+            memory_slot (int): Number of memory slot to load model to. Valid values are 1 to 10.
+            model_file_name (str): Name of battery model to load. Omit '.csv'
+
+        Raises:
+            ValueError: invalid memory slot given
+        """
+        if _INTERNAL_MEMORY_SLOTS[0] > memory_slot > _INTERNAL_MEMORY_SLOTS[-1]:
+            raise ValueError
+        self.write(f":BATT:MOD:LOAD:USB {memory_slot}, \"{model_file_name}\"")
+
+    output_enabled = Instrument.control(
+        ":BATT:OUTP?",
+        ":BATT:OUTP %s",
+        """Control the output state.""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    simulation_mode = Instrument.control(
+        ":BATT:SIM:METH?",
+        ":BATT:SIM:METH %s",
+        """
+        Control simulation mode to use.
+        I.e. does the SoC change when charging or discharging.
+        """,
```

Comment:
It is good to add the possible values to the docstring.
Do I understand it correctly, that there are only two possible modes?
What do you think about a boolean property `dynamic_simulation_enabled`? That makes it easier to use and to interpret as it is just a boolean and not strings being sent to the device/interpreted.